zstacklib/zstacklib/test/test_uribuilder.py

```
@@ -38,7 +38,34 @@ def test_build_url5(self):
+        self.assertEqual('http://[fd11:5:5:29::220]:80/zstack/api/', builder.build())
+
+    def test_build_url_with_ipv6_uri_default_https_port(self):
+        builder = http.UriBuilder('https://[fd11:5:5:29::220]/zstack/api/')
+        self.assertEqual('https://[fd11:5:5:29::220]:443/zstack/api/', builder.build())
+
+    def test_build_url_with_https_default_port(self):
+        builder = http.UriBuilder('https://localhost/')
+        self.assertEqual('https://localhost:443/', builder.build())
+
+    def test_build_url_with_invalid_port(self):
+        self.assertRaises(Exception, http.UriBuilder, 'http://localhost:abc/')
```

Ruff (0.15.18)

[warning] Do not assert blind exception: `Exception`

(B017)